spec/models/on_duty_locator_spec.rb

```diff
@@ -10,102 +10,164 @@
 
   let(:defined_shift) {
     build_stubbed :shift,
-                        starting_time: Time.parse("08:00"),
-                        ending_date: Time.parse("16:00")
+    starting_time: Time.parse("08:00"),
+    ending_date: Time.parse("16:00")
   }
 
-  it "returns the single uid for the organisation on duty" do
-    time = Time.parse("01/01/2014 20:00").iso8601
+  context "when one firm is on duty" do
+    it "returns the single uid for the organisation" do
+      time = Time.parse("01/01/2014 20:00").iso8601
 
-    create(:rota_slot,
-           organisation_uid: firm_1,
-           starting_time: Time.parse("01/01/2014 12:00"),
-           shift: first_shift)
+      create(:rota_slot,
+             organisation_uid: firm_1,
+             starting_time: Time.parse("01/01/2014 12:00"),
+             shift: first_shift)
 
-    create(:rota_slot,
-           organisation_uid: firm_2,
-           starting_time: Time.parse("01/01/2014 21:00"),
-           shift: last_shift)
+      create(:rota_slot,
+             organisation_uid: firm_2,
+             starting_time: Time.parse("01/01/2014 21:00"),
+             shift: last_shift)
 
-    on_duty_organisation_uids = OnDutyLocator.new(time, RotaSlot.order(starting_time: :desc)).locate
+      on_duty_organisation_uid = OnDutyLocator.new(time, RotaSlot.order(starting_time: :desc)).locate
 
-    expect(on_duty_organisation_uids).to eq [firm_1]
-  end
+      expect(on_duty_organisation_uid).to eq firm_1
+    end
 
-  it "returns multiple uids for the organisations on duty" do
-    time = Time.parse("01/01/2014 20:00").iso8601
+    it "updates the request count for that slot" do
+      time = Time.parse("01/01/2014 20:00").iso8601
 
-    create(:rota_slot,
-           organisation_uid: firm_1,
-           starting_time: Time.parse("01/01/2014 12:00"),
-           shift: first_shift)
+      rota_slot = create(:rota_slot,
+                         organisation_uid: firm_1,
+                         starting_time: Time.parse("01/01/2014 12:00"),
+                         shift: first_shift,
+                         request_count: 6)
 
-    create(:rota_slot,
-           organisation_uid: firm_2,
-           starting_time: Time.parse("01/01/2014 12:00"),
-           shift: first_shift)
+      expect {
+        OnDutyLocator.new(time, RotaSlot.order(starting_time: :desc)).locate
+      }.to change {
+        rota_slot.reload.request_count
+      }.by(1)
```

Comment:
Be good to use the ```.from(x).to(y)``` expectations to set the start and end values as this removes ambiguity in the test